ziti/src/main/kotlin/org/openziti/net/nio/SyncSocketChannel.kt

```diff
@@ -0,0 +1,193 @@
+        validateConnection()
+        return channelImpl.write(src).get()
+    }
+
+    override fun write(srcs: Array<out ByteBuffer>?, offset: Int, length: Int): Long {
+        validateConnection()
+        var response: Long
+
+        runBlocking {
+            response = suspendCoroutine { cont ->
+                channelImpl.write(
+                    srcs,
+                    offset,
+                    length,
+                    10000,
+                    TimeUnit.MILLISECONDS,
+                    this,
+                    object : CompletionHandler<Long, CoroutineScope> {
+                        override fun completed(result: Long?, attachment: CoroutineScope?) {
+                            cont.resume(result ?: -1)
+                        }
+
+                        override fun failed(exc: Throwable?, attachment: CoroutineScope?) {
+                            if (null != exc) cont.resumeWithException(exc)
+                            else cont.resumeWithException(IllegalStateException("Channel read failed without an exception"))
+                        }
+                    })
+            }
+        }
+        return response
+    }
+
+    override fun bind(local: SocketAddress?): SocketChannel = error("only client sockets are supported")
+
+    override fun getLocalAddress(): SocketAddress {
+        validateConnection()
+        return channelImpl.localAddress
+    }
+
+    override fun <T : Any?> setOption(name: SocketOption<T>?, value: T): SocketChannel {
+        validateConnection()
+        channelImpl.setOption(name, value)
+        return this
+    }
+
+    override fun <T : Any?> getOption(name: SocketOption<T>?): T {
+        validateConnection()
+        return channelImpl.getOption(name)
+    }
+
+    override fun supportedOptions(): MutableSet<SocketOption<*>> {
+        validateConnection()
+        return channelImpl.supportedOptions()
+    }
+
+    override fun shutdownInput(): SocketChannel {
+        validateConnection()
+        channelImpl.shutdownInput()
+        return this
+    }
+
+    override fun shutdownOutput(): SocketChannel {
+        validateConnection()
+        channelImpl.shutdownOutput()
+        return this
+    }
+
+    override fun socket(): Socket {
+        validateConnection()
+        return AsyncChannelSocket(channelImpl)
+    }
+
+    override fun isConnected(): Boolean = this::channelImpl.isInitialized && channelImpl.isOpen
+
+    override fun isConnectionPending(): Boolean = this::channelImpl.isInitialized && !channelImpl.isOpen
+
+    override fun connect(addr: SocketAddress?): Boolean {
+        val sockAddr = addr as InetSocketAddress
+        for (ctx in Ziti.getContexts()) {
+            val ctxImpl = ctx as ZitiContextImpl
+            try {
+                val s = ctxImpl.getService(sockAddr)
+                if (s != null) {
+                    channelImpl = ZitiSocketFactory.ZitiConnector.doConnect(ctx.open(), sockAddr, 10000)
```

Comment:
why not `ctx.open().connect(sockAddr)`?

Reply:
I didn't know that was an option.  I'll update the code